src/main/java/seedu/address/logic/commands/SetDefaultGroupCommand.java

```diff
@@ -40,16 +41,15 @@ public SetDefaultGroupCommand(Mod mod, String defaultValue) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        UniqueModuleList moduleList = new UniqueModuleList();
         if (model.doesModExist(mod)) {
-            Mod matchingMod = moduleList.retrieveMod(mod).get(); //since guaranteed to not be null
-            if (model.isDefaultPresent(matchingMod)) {
-                String prevDefault = model.retrievePrevDefault(matchingMod);
+            Optional<Mod> matchingMod = model.getMod(mod); //since guaranteed to not be null
```

Comment:
I think you can include an `assert matchingMod != null` here for defensive programming, what do you think?